### PR TITLE
Update falco 20230215

### PR DESCRIPTION
**Pull request by Stringy**
Experimental attempt at merging rather than rebasing...

### CODE REVIEW

File: userspace/libscap/linux/scap_procs.c [hunk 1]
```diff
@@ -1030,7 +1054,7 @@ static int32_t scap_proc_add_from_proc(scap_t* handle, uint32_t tid, char* procd
 	//
 	if(tinfo->pid == tinfo->tid)
 	{
-		/* Begin StackRox Section */
+        /* Begin StackRox Section */
```
Comment (Molter73) [inline userspace/libscap/linux/scap_procs.c:1057]: [ultra-nit] Another weird indentation change

File: userspace/libscap/linux/scap_procs.c [hunk 2]
```diff
@@ -1245,7 +1260,7 @@ struct scap_threadinfo* scap_proc_get(scap_t* handle, int64_t tid, bool scan_soc
 	if(scap_proc_read_thread(handle, filename, tid, &tinfo, handle->m_lasterr, scan_sockets) != SCAP_SUCCESS)
 	{
 		free(tinfo);
-		/* Begin StackRox Section */
+        /* Begin StackRox Section */
```
Comment (Molter73) [inline userspace/libscap/linux/scap_procs.c:1263]: [ultra-nit] Another weird indentation change

File: userspace/libsinsp/container_engine/cri.cpp [hunk 3]
```diff
@@ -76,7 +76,7 @@ bool cri_async_source::parse_containerd(const runtime::v1alpha2::ContainerStatus
 			container.m_id.c_str(),
 			info_it->second.c_str());
 
-	/* Begin StackRox - Image labels and env vars are not used by StackRox collector (ROX-6200) */
+    /* Begin StackRox - Image labels and env vars are not used by StackRox collector (ROX-6200) */
```
Comment (Molter73) [inline userspace/libsinsp/container_engine/cri.cpp:79]: [ultra-nit] Another weird indentation change

**Review by Molter73 — COMMENTED** [review]
Looks good! I've not looked much into the changes to the "modern" probe, @erthalion maybe you can go through those to ensure we have everything we need for BTF enablement?

I marked a few weird indentation changes, these are very minor, but they could trigger new conflicts with future commits, so it'd be nice to roll them back if at all possible.

File: cmake/modules/grpc.cmake [hunk 4]
```diff
@@ -41,8 +41,8 @@ elseif(NOT USE_BUNDLED_GRPC)
 			set(GRPC_INCLUDE ${GRPCPP_INCLUDE})
 			add_definitions(-DGRPC_INCLUDE_IS_GRPCPP=1)
 		endif()
-		find_library(GRPC_LIB NAMES libgrpc.a grpc)
-		find_library(GRPCPP_LIB NAMES libgrpc++.a grpc++)
+		find_library(GRPC_LIB NAMES grpc)
+		find_library(GRPCPP_LIB NAMES grpc++)
```
[the remark this answers is not in the document]
Comment (Stringy) [reply cmake/modules/grpc.cmake:45]: But aren't we disabling gRPC linking in collector CMakeLists? https://github.com/stackrox/collector/blob/master/collector/CMakeLists.txt#L99

File: driver/bpf/probe.c [hunk 5]
```diff
@@ -50,18 +49,11 @@ BPF_PROBE("raw_syscalls/", sys_enter, sys_enter_args)
...
 	{
 		return 0;
 	}
 
-	settings = get_bpf_settings();
-	if (!settings)
-		return 0;
-
-	if (!settings->capture_enabled)
-		return 0;
```
[the remark this answers is not in the document]
Comment (Stringy) [reply driver/bpf/probe.c:63]: This change is also replicated in our custom probe (see stackrox/collector#1021) but I think the performance impact will be very very small, especially in our case where we're only executing on syscalls we want to capture anyway.

File: cmake/modules/grpc.cmake (hunk 4 above)
[the remark this answers is not in the document]
Comment (Molter73) [reply cmake/modules/grpc.cmake:45]: You are right, never mind this comment.

**Review by Molter73 — COMMENTED** [review]
Other than restoring the link to `libre2.a`, everything else LGTM!

File: userspace/libsinsp/threadinfo.cpp [hunk 6]
```diff
@@ -1362,7 +1411,7 @@ void sinsp_thread_manager::increment_mainthread_childcount(sinsp_threadinfo* thr
 		//
 		ASSERT(threadinfo->m_pid != threadinfo->m_tid);
 
-		sinsp_threadinfo* main_thread = m_inspector->get_thread_ref(threadinfo->m_pid, create_if_needed, true).get();
+		sinsp_threadinfo* main_thread = m_inspector->get_thread_ref(threadinfo->m_pid, true, true).get();
```
Comment (Molter73) [inline userspace/libsinsp/threadinfo.cpp:1414]: I'm pretty sure we want to keep this as `create_if_needed`, this is one of our changes.

**Review by Molter73 — COMMENTED** [review]
Just one minor comment, other than that I think we are good to merge

**Pull request merged**